nix/stack-shell.nix

```diff
@@ -0,0 +1,33 @@
+
+# This shell file is specifically to be used with Stack.
+#
+# This file allows using Stack's built-in Nix integration.  This means that you
+# can compile hasktorch with Stack by using a command like `stack --nix build`.
+# Stack will use Nix to download and build required system libraries (like GHC
+# and libtorch), and then build Haskell libraries like normal.
+#
+# This approach allows for more reproducibility than using Stack without Nix.
+
+{ withHoogle ? false
+}@args:
+
+let
+  shared = import ../shared.nix {};
+  pkgs = shared.defaultCpu.pkgs;
+in
+
+with pkgs;
```

Comment:
I was able to simplify this by just pulling out `pkgs` from the `shared.nix` file.

Again, this should eventually be updated to not always use `defaultCpu`, but take an cuda argument to figure out whether to use non-cuda cpu, or cuda.  However, I think this could be added in a future PR.